var/spack/repos/builtin/packages/qmcpack/package.py

```diff
@@ -92,6 +97,16 @@ class Qmcpack(CMakePackage, CudaPackage):
     conflicts('%pgi@:17', when='@3.6.0:', msg=compiler_warning)
     conflicts('%llvm@:3.4', when='@3.6.0:', msg=compiler_warning)
 
+    conflicts('+afqmc', when='@:3.6.0', msg='AFQMC not recommended before v3.7')
+    conflicts('+afqmc', when='~mpi', msg='AFQMC requires building with +mpi')
+    conflicts('+afqmc', when='~complex +cuda', msg='AFQMC real build not supported with CUDA. Add +complex')
+    conflicts('+afqmc', when='%gcc@:6.0', msg='AFQMC code requires gcc@6.1 or greater')
+    conflicts('+afqmc', when='%clang@:4.0', msg='AFQMC code requires clang 4.1 or greater')
+    conflicts('+afqmc', when='%intel@:18',msg='AFQMC code requires intel19 or greater')
```

Comment:
Could you add a comment about why these version of the compiler are needed? I am assuming this is due to C++17 support required to compile AFQMC.

Also, do you know about PGI conflicts?

Reply:
Personally, I haven't tested these. I posted an issue on the qmcpack github (issue 2237)  , and asked about any known conflicts.  Fionn mentioned stating these compiler versions are known to work and are currently tested. So I added the conflict in case there are any issues with earlier versions. 

I do not know about PGI conflicts. I can ask Fionn